gog.js

```diff
@@ -136,17 +159,141 @@ try {
+        console.log("Play For Free. Can't be added to library!" + url);
+        return;
+      }
+
+      await page.waitForURL('**/checkout/**');
+      if (await page.locator('.order-message--error').isVisible()) {
+        console.log("skipping : " + await page.locator('.order-message--error').innerText());
+        await page.locator('span[data-cy="product-remove-button"]').click();
+        return;
+      }
+
+      await page.locator('button[data-cy="payment-checkout-button"]').click();
+
+      await page.waitForURL('**/order/status/**');
+      await page.locator('p[data-cy="order-message"]').isVisible();
+      
+      notify_game.status = "claimed";
+      db.data[user][game_id].status = "claimed";
+      db.data[user][game_id].time = datetime(); // claimed time overwrites failed/dryrun time
+      console.log("Claimed successfully!");
+      await db.write();
+    }
+
+    const p = path.resolve(cfg.dir.screenshots, 'gog', `${game_id}.png`);
+    if (!existsSync(p)) await page.screenshot({ path: p, fullPage: false }); // fullPage is quite long...
 }
-if (page.video()) console.log('Recorded video:', await page.video().path())
+
+async function claimFreegames(){
+  console.log("claiming freegames from " + cfg.gog_freegames_url + ". (adding fitler for ownedgames manually)")
+  await page.goto(cfg.gog_freegames_url, { waitUntil: 'networkidle' });
+  await page.locator('label[selenium-id="hideOwnedCheckbox"]').click(); // when you add it to url immediately it shows more results
+  await page.waitForTimeout(2500);
+  var allLinks = [];
+  var hasMorePages = true;
+  do {
+    const links = await page.locator(".product-tile").all()
+    const gameUrls = await Promise.all(
+        links.map(async (game) => {
+          var urlSlug = await game.getAttribute("href");
+          return urlSlug;
+      })
+    );
+    for (const url of gameUrls) {
+      allLinks.push(url);
+    }
+    if (await page.locator('.small-pagination__item--next.disabled').isVisible()){
+      hasMorePages = false
+      console.log("last page")
+    } else {
+      await page.locator(".small-pagination__item--next").first().click();
+      console.log("next page - waiting")
+      await page.waitForTimeout(5000); // wait until page is loaded it takes some time with filters
+    }
+    
+  } while (hasMorePages)
+  console.log("Found total games: " + allLinks.length)
+  allLinks = allLinks.filter(function (str) { return !str.endsWith("_prologue") });
+  allLinks = allLinks.filter(function (str) { return !str.endsWith("_demo") });
+  console.log("Filtered count: " + allLinks.length)
+
+  for (const url of allLinks)
+    {
+        if (isNotClaimedUrl(url))
+        {
+          console.log(url)
+          await claimGame(url);
+        }
+    }
+}
+
+function isNotClaimedUrl(url) {
+    try {
+        var status = db.data[user][url.split("/").filter((x) => !!x).pop()]["status"];
+        if (status === "existed" || status === "claimed") {
```

Comment:
Why not just return the negated condition?
Also, some statements are ended with `;` and some are not.

Reply:
I meant replace the if-then-else with
```js 
return status === "existed" || status === "claimed";
```